VM/src/lvmexecute.cpp

```diff
@@ -320,6 +332,8 @@ static void luau_execute(lua_State* L)
     base = L->base;
     k = cl->l.p->k;
 
+    signal(SIGINT, handle_sig);
```

Comment:
Correction: it should be sufficient for this entire feature to be implemented in Repl; instead of setting a status variable (and checking it in VM_INTERRUPT), it should be sufficient to dynamically set the interrupt callback from Repl. Some care needs to be taken in the profiler support code (in Repl) so that the profile callback doesn't overwrite the interrupt callback, but otherwise the interrupt callback is a fully general mechanism that should be sufficient to perform termination here.